### DISCUSSION TITLE
List of repositories of the selected user (username string) has contributed (pull req, commit, etc)

### DISCUSSION THREAD

**Discussion by efendyklb** [community/community#24440] [Apps, API and Webhooks]
<p>Hello everyone,</p>
<p>i am new for the graphql, my project is I want to build some repository dashboard in my school</p>
<p>my case is i want to my friend’s public repositories that he/she has contributed to, sorted by the most recent contribution date.</p>
<p>in my case I stuck in query like this :</p>
<pre><code>{
  search (query: "Effzz", type: USER, first: 1){
    edges {
      node {
        ... on User {
          login
          url
          avatarUrl
          bio
          repositories(first: 10 orderBy: {field: UPDATED_AT, direction: DESC}) {
            pageInfo {
              endCursor
              startCursor
              hasNextPage
              hasPreviousPage
            }
            edges {
              node {
                name
                id
                createdAt
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>but this just showing last updated user’s own repository,</p>
<p>any single advice is really appreciated,</p>
<p>thank you</p>

**Comment by TimonVS** [comment, marked as answer]
<p>Hi!</p>
<p>Would this be what you’re looking for? This query returns the contributions from a user grouped per repository. You might still have to do some postprocessing on the data. If this is not what you’re looking for <code>contributionsCollection</code> might still contain some other fields that can be of interest to you.</p>
<pre><code>{
  user(login: "TimonVS") {
    login
    contributionsCollection {
      commitContributionsByRepository {
        contributions(last: 10) {
          nodes {
            occurredAt
            commitCount
            repository {
              nameWithOwner
            }
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by efendyklb** [comment]
<p>thanks sir, this is what I need, gracias</p>
